chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -472,6 +477,40 @@ impl PeerManagerActor {
             .collect()
     }
 
+    // Go over all the peers that you're connected to, and refresh the connection nonce if needed.
+    // We only have to do it for the outgoing edges (as for incoming we assume that the other node is going to this).
+    fn refresh_peer_nonces(&self) {
+        let tier2 = self.state.tier2.load();
+        let peers_to_refresh = tier2
+            .ready
+            .values()
+            .filter(|connection| {
+                connection
+                    .edge
+                    .load()
+                    .is_edge_older_than(self.clock.now_utc() - REFRESH_NONCE_PERIOD)
+                    && connection.peer_type == PeerType::Outbound
+            })
+            .collect::<Vec<_>>();
+        if peers_to_refresh.len() > 0 {
+            debug!(target:"network", "Refreshing nonces for {:?} peers.", peers_to_refresh.len());
```

Comment:
this logic should be moved inside PeerActor IMO. Let every actor take care of their edge on their own.

Reply:
done